tests/testsuite/resolve.rs

```diff
@@ -185,6 +190,97 @@ fn loc_names(names: &[(&'static str, &'static str)]) -> Vec<PackageId> {
         .collect()
 }
 
+/// This generates a random registry index.
+/// Unlike vec((Name, Ver, vec((Name, VerRq), ..), ..)
+/// This strategy has a high probability of having valid dependencies
+fn registry_strategy() -> impl Strategy<Value=Vec<Summary>> {
+    const VALID_NAME_STRATEGY: &str = "[A-Za-z_-][A-Za-z0-9_-]*";
+    const MAX_CRATES: usize = 10;
+    const MAX_VERSIONS: usize = 10;
+
+    fn range(max: usize) -> impl Strategy<Value = (usize, usize)> {
+        (0..max).prop_flat_map(move |low| (Just(low), low..=max))
```

Comment:
I believe you could rewrite this as

```rust
(0..max, 0..max).prop_map(|(a, b)| min(a, b)..=max(a,b))
```

which could substantially improve shrinking performance